site/tests/inspire_harvester/test_writer.py

```diff
@@ -0,0 +1,717 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2025 CERN.
+#
+# CDS-RDM is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+"""ISNPIRE harvester writer tests."""
+
+import pytest
+from cds_rdm.inspire_harvester.writer import InspireWriter
+from invenio_access.permissions import system_identity
+from invenio_rdm_records.proxies import (current_rdm_records,
+                                         current_rdm_records_service)
+from invenio_vocabularies.datastreams import StreamEntry
+from invenio_vocabularies.datastreams.errors import WriterError
+
+
+def _cleanup_record(recid):
+    """Delete a record after each test."""
+    current_rdm_records.records_service.delete(system_identity, recid)
+
+
+@pytest.fixture()
+def transformed_record_1_file():
+    """Transformed via InspireJsonTransformer record with 1 file."""
+    return {
+        "metadata": {
```

Comment:
I understand the reader is not covered with any tests, we don't know if pagination would work etc. Do you mind adding a test which is mocking the Inspire response to test the whole workflow with almost real response object?